src/neptune/internal/backends/hosted_neptune_backend.py

```diff
@@ -834,10 +834,11 @@ def get_datetime_attribute(
     def get_artifact_attribute(
         self, container_id: str, container_type: ContainerType, path: List[str]
     ) -> ArtifactAttribute:
+        requests_params = self._add_artifact_version_to_request_params(DEFAULT_REQUEST_KWARGS)
         params = {
             "experimentId": container_id,
             "attribute": path_to_str(path),
-            **DEFAULT_REQUEST_KWARGS,
+            **requests_params,
         }
         try:
             result = self.leaderboard_client.api.getArtifactAttribute(**params).response().result
```

Comment:
How about delegating this method to `hosted_artifact_operations`? I mean moving the content to some method that accepts swagger API object.

Reply:
Do you mean to add `X-Neptune-Artifact-Api-Version` to all requests within `hosted_artifact_operations`? 
Initially, I did it like that, but later notice we used it only with `get_artifact_attribute`.

Are there any plans for that header in the future, that I'm not aware of?

Reply:
Done. Also moved `list_artifact_files` method implementation.